### PR TITLE
PR10 - Remove AlicaEngine::this from RuntimeBehaviourFactory 

### PR DESCRIPTION
**Pull request by triccyx**
PR10 - Remove AlicaEngine::this from RuntimeBehaviourFactory  and subclasses

### CODE REVIEW

File: alica_engine/include/engine/RuntimeBehaviourFactory.h [hunk 1]
```diff
@@ -11,15 +11,19 @@ class Behaviour;
 class BasicBehaviour;
 class IBehaviourCreator;
 class IAlicaWorldModel;
-
```
Comment (christianhelm) [inline alica_engine/include/engine/RuntimeBehaviourFactory.h:14]: Forward declaration of AlicaEngine can be removed now

File: alica_engine/src/engine/RunnableObject.cpp [hunk 2]
```diff
@@ -13,7 +13,6 @@ namespace alica
 {
 RunnableObject::RunnableObject(IAlicaWorldModel* wm, const std::string& name)
         : _name(name)
```
Comment (christianhelm) [inline alica_engine/src/engine/RunnableObject.cpp:16]: Shouldn't need to include AlicaEngine.h anymore

File: alica_engine/src/engine/BasicPlan.cpp [hunk 3]
```diff
@@ -62,7 +62,7 @@ void BasicPlan::doTerminate()
 
 void BasicPlan::traceAssignmentChange(const std::string& assignedEntryPoint, double oldUtility, double newUtility, size_t numberOfAgents)
 {
-    if (_engine->getTraceFactory() && getTrace()) {
+    if (_traceFactory && getTrace()) {
```
Comment (christianhelm) [inline alica_engine/src/engine/BasicPlan.cpp:64]: Shouldn't need to include AlicaEngine.h anymore

File: alica_engine/src/engine/BasicBehaviour.cpp [hunk 4]
```diff
@@ -46,7 +46,7 @@ BasicBehaviour::BasicBehaviour(BehaviourContext& context)
  */
 AgentId BasicBehaviour::getOwnId() const
 {
-    return _engine->getTeamManager().getLocalAgentID();
```
Comment (christianhelm) [inline alica_engine/src/engine/BasicBehaviour.cpp:49]: Shouldn't need to include AlicaEngine.h anymore

File: alica_engine/src/engine/RuntimeBehaviourFactory.cpp [hunk 5]
```diff
@@ -30,7 +35,11 @@ std::unique_ptr<BasicBehaviour> RuntimeBehaviourFactory::create(int64_t id, cons
     }
 
     // TODO Cleanup: get rid of this later, behaviour only needs traceFactory, teamManager and not entire engine
-    basicBeh->setEngine(_engine);
+    basicBeh->setTeamManager(&_teamManager);
```
Comment (Dale-Koenig) [inline alica_engine/src/engine/RuntimeBehaviourFactory.cpp:37]: Remove Todo?

File: alica_engine/src/engine/RuntimePlanFactory.cpp [hunk 6]
```diff
@@ -31,7 +31,10 @@ std::unique_ptr<BasicPlan> RuntimePlanFactory::create(int64_t id, const Plan* pl
     }
 
     // TODO Cleanup: get rid of this later, behaviour only needs traceFactory, teamManager and not entire engine
-    basicPlan->setEngine(_engine);
+    basicPlan->setAlicaTraceFactory(_engine->getTraceFactory());
```
Comment (Dale-Koenig) [inline alica_engine/src/engine/RuntimePlanFactory.cpp:34]: Remove Todo

File: alica_engine/src/engine/allocationauthority/CycleManager.cpp [hunk 7]
```diff
@@ -27,21 +27,19 @@ using std::mutex;
  * @param p A RunningPlan
  */
 CycleManager::CycleManager(ConfigChangeListener& configChangeListener, const AlicaClock& clock, const TeamManager& teamManager,
-        const PlanRepository& planRepository, RunningPlan* rp)
+        const PlanRepository& planRepository, RunningPlan* p)
```
Comment (Dale-Koenig) [inline alica_engine/src/engine/allocationauthority/CycleManager.cpp]: Why the rename

File: alica_tests/src/test/test_task_assignment.cpp [hunk 8]
```diff
@@ -145,6 +145,7 @@ TEST_F(TaskAssignmentTest, switchEntryPoints)
             ae->getPlanRepository().getPlans().find(1407152758497), nullptr);
 
     // fake inform the team observer about roles of none existing robots
+    // alica::RunningPlan* rp = new RunningPlan(ae, ae->getPlanRepository().getPlans().find(1407152758497), nullptr);
```
Comment (Dale-Koenig) [inline alica_tests/src/test/test_task_assignment.cpp]: Remove

File: alica_tests/src/test/test_config_change.cpp [hunk 9]
```diff
@@ -93,4 +93,4 @@ TEST_F(AlicaNotInitialized, TestConfigUpdatesWithVector)
     EXPECT_EQ(1000.0f, ac->getConfig()["Local"]["AverageTranslation"].as<float>());
 }
 } // namespace
-} // namespace alica
+} // namespace alica
```
Comment (Dale-Koenig) [inline alica_tests/src/test/test_config_change.cpp]: Configure editor to always have newline at eof

File: alica_engine/include/engine/RuntimeBehaviourFactory.h (hunk 1 above)
Comment (triccyx) [reply alica_engine/include/engine/RuntimeBehaviourFactory.h:14]: ok

File: alica_engine/src/engine/BasicPlan.cpp (hunk 3 above)
Comment (triccyx) [reply alica_engine/src/engine/BasicPlan.cpp:64]: ok

File: alica_engine/src/engine/BasicBehaviour.cpp (hunk 4 above)
Comment (triccyx) [reply alica_engine/src/engine/BasicBehaviour.cpp:49]: ok

File: alica_engine/src/engine/RuntimeBehaviourFactory.cpp (hunk 5 above)
Comment (triccyx) [reply alica_engine/src/engine/RuntimeBehaviourFactory.cpp:37]: ok

File: alica_engine/src/engine/allocationauthority/CycleManager.cpp (hunk 7 above)
Comment (triccyx) [reply alica_engine/src/engine/allocationauthority/CycleManager.cpp]: ok

File: alica_engine/src/engine/RuntimePlanFactory.cpp (hunk 6 above)
Comment (triccyx) [reply alica_engine/src/engine/RuntimePlanFactory.cpp:34]: still needed until next PR11

**Comment by triccyx** [comment]
Remove all (I hope) unused include and fw declaration to AlicaEngine around

**Review by christianhelm — APPROVED** [review]

**Review by Dale-Koenig — APPROVED** [review]

**Pull request merged**